pkg/cache/tas_flavor_snapshot.go

```diff
@@ -296,19 +297,34 @@ func (s *TASFlavorSnapshot) updateCountsToMinimum(domains []*domain, count int32
 }
 
 func (s *TASFlavorSnapshot) buildAssignment(domains []*domain) *kueue.TopologyAssignment {
-	assignment := kueue.TopologyAssignment{
-		Levels:  s.levelKeys,
-		Domains: make([]kueue.TopologyDomainAssignment, 0),
-	}
+	assignment := kueue.TopologyAssignment{}
 	// lex sort domains
 	slices.SortFunc(domains, func(a, b *domain) int {
 		return cmp.Compare(a.id, b.id)
 	})
-	for _, domain := range domains {
-		assignment.Domains = append(assignment.Domains, kueue.TopologyDomainAssignment{
-			Values: domain.levelValues,
-			Count:  domain.state,
-		})
+	// assign only hostname values if topology defines it
+	if idx, found := s.resolveLevelIdx(corev1.LabelHostname); found {
+		assignment = kueue.TopologyAssignment{
+			Levels:  []string{corev1.LabelHostname},
+			Domains: make([]kueue.TopologyDomainAssignment, len(domains)),
+		}
+		for i, domain := range domains {
+			assignment.Domains[i] = kueue.TopologyDomainAssignment{
+				Values: []string{domain.levelValues[idx]},
+				Count:  domain.state,
+			}
+		}
+	} else {
+		assignment = kueue.TopologyAssignment{
+			Levels:  s.levelKeys,
+			Domains: make([]kueue.TopologyDomainAssignment, len(domains)),
+		}
+		for i, domain := range domains {
+			assignment.Domains[i] = kueue.TopologyDomainAssignment{
+				Values: domain.levelValues,
+				Count:  domain.state,
+			}
+		}
```

Comment:
I would like to commonize the code between the two blocks, but no concrete suggestion yet. Maybe a helper function which serializes the assignments starting from index?

Reply:
Done